astropy/cosmology/_src/core.py

```diff
@@ -203,19 +207,19 @@ def _register_cls(cls):
 
     # ---------------------------------------------------------------
 
-    def __post_init__(self):  # noqa: B027
+    def __post_init__(self) -> None:  # noqa: B027
```

Comment:
I don't understand the point of defining this method as no-op.

Reply:
I see. But then, what's the point of having this ? I would argue that most (data)classes should not need inheritance at all, so I think I would rather have it be slightly more difficult to call `super().__post_init__` *when* it's actually needed (via a `try/except AttributeError`) than having "just in case" code here. And if you're unconvinced, I would still advise to add an inline comment here.